app/models/cloud_tenant.rb

```diff
@@ -14,6 +14,7 @@ class CloudTenant < ApplicationRecord
   has_many   :network_routers
   has_many   :vms
   has_many   :vms_and_templates
+  has_many   :active_vms, -> { active }, :foreign_key => :cloud_tenant_id, :class_name => "Vm"
```

Comment:
might be that we just need ```has_mane :vms, -> { active }```

it's consistent with the behavior elsewhere, where we delete e.g. availability_zone_id. Seems like we've forgot to delete the cloud_tenant_id here. So ```has_mane :vms, -> { active }``` is equivalent to having cloud_tenant_id deleted, but it's cleaner because we can keep the foreign key.